mozetl/sync/bookmark_validation.py

```diff
@@ -0,0 +1,133 @@
+       s.when,
+       s.status,
+       e.name AS engine_name,
+       e.status AS engine_status,
+       e.failureReason AS engine_failure_reason,
+       e.validation.problems IS NOT NULL AS engine_has_problems,
+       e.validation.version AS engine_validation_version,
+       e.validation.checked AS engine_validation_checked,
+       e.validation.took AS engine_validation_took,
+       p.name AS engine_validation_problem_name,
+       p.count AS engine_validation_problem_count
+    FROM sync_summary s
+    LATERAL VIEW explode(s.engines) AS e
+    LATERAL VIEW OUTER explode(e.validation.problems) AS p
+    WHERE s.failureReason IS NULL
+    """
+    engine_validation_results = spark.sql(query)
+
+    bmk_validation_results = (
+        engine_validation_results
+        .filter(engine_validation_results["engine_name"] == "bookmarks")
+    )
+
+    # Bookmark validations with problems.
+    bmk_validation_problems = (
+        bmk_validation_results
+        .filter(bmk_validation_results["engine_has_problems"])
+    )
+
+    # All bookmark validations, including without problems.
+    bmk_total_per_day = (
+        bmk_validation_results
+        .filter(bmk_validation_results["engine_validation_checked"].isNotNull())
+        .groupBy("sync_day")
+        .agg(
+            F.countDistinct("uid", "device_id", "when").alias("total_bookmark_validations"),
+            F.countDistinct("uid").alias("total_validated_users"),
+            F.sum("engine_validation_checked").alias("total_bookmarks_checked")
+        )
+    )
+
+    bmk_validation_problems.createOrReplaceTempView("bmk_validation_problems")
+    bmk_total_per_day.createOrReplaceTempView("bmk_total_per_day")
+
+
+def load(spark, bucket, prefix, version, start_date):
+    """Save tables to disk."""
+
+    for table_name in ["bmk_validation_problems", "bmk_total_per_day"]:
+        path = (
+            "s3://{}/{}/{}/v{}/start_date={}"
+            .format(bucket, prefix, table_name, version, start_date)
+        )
+
+        logger.info("Saving table {} on start_date {} to {}"
+                    .format(table_name, start_date, path))
+
+        df = spark.sql("SELECT * FROM {}".format(table_name))
+        df.repartition(1).write.parquet(path, mode="overwrite")
+
+
+@click.command()
+@click.option('--start_date', required=True, help="Date to process")
+@click.option('--end_date', help="Optional end date to run until")
+@click.option('--bucket', default='net-mozaws-prod-us-west-2-pipeline-analysis')
+@click.option('--prefix', default='kit/sync')
+@click.option('--input_bucket', default='telemetry-parquet')
+@click.option('--input_prefix', default='sync_summary/v1')
+def main(start_date, end_date, bucket, prefix, input_bucket, input_prefix):
+    spark = (SparkSession
+             .builder
+             .appName("sync_bookmark")
+             .getOrCreate())
+
+    # configure the spark session to partition the data reasonably
```

Comment:
I don't think this block size configuration has any effect (I think it's a holdover from old versions). Also, the `load` function collapses everything into 1 partition, so only one file will be written per table/`start_date` regardless of these block size directives. It should be safe to remove these lines.